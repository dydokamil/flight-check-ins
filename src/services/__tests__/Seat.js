import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'

import {
  mapStateToProps,
  mapDispatchToProps,
  Seat,
} from '../../components/Seat'

Enzyme.configure({ adapter: new Adapter() })

describe('test Seats component', () => {
  const INITIAL_STATE = {
    seat: { id: 1 },
    checkInFee: 0,
    basePrice: 0,
    onBookSeat: jest.fn(),
    sessionReducer: {
      token: '123abc',
      email: 'john@doe.com',
    },
  }

  it('should render the component', () => {
    const component = shallow(<Seat {...INITIAL_STATE} />)

    expect(component).toMatchSnapshot()
  })
  it('should assign email prop a value from a reducer', () => {
    expect(mapStateToProps(INITIAL_STATE)).toEqual({
      token: INITIAL_STATE.sessionReducer.token,
      email: INITIAL_STATE.sessionReducer.email,
    })
  })

  it('should call `onBookSeat` when a seat is booked', () => {
    const dispatch = jest.fn()
    mapDispatchToProps(dispatch).onBookSeat({})
    expect(dispatch).toHaveBeenCalledTimes(1)
  })

  it('executes `toggle` on button click', () => {
    const wrapper = shallow(<Seat {...INITIAL_STATE} />)

    wrapper.find(`#Popover${INITIAL_STATE.seat.id}`).simulate('click')
    expect(wrapper.instance().state.popoverOpen).toBeTruthy()
  })

  it('executes `toggle` on book', () => {
    const wrapper = shallow(<Seat {...INITIAL_STATE} />)

    wrapper.find(`#Popover${INITIAL_STATE.seat.id}`).simulate('click')
    setTimeout(() => {
      wrapper.find("button[children='Book']").simulate('click')
      expect(wrapper.instance().state.popoverOpen).toBeTruthy()
      expect(INITIAL_STATE.onBookSeat).toHaveBeenCalled()
      expect(wrapper.instance().state.popoverOpen).toBeFalsy()
    }, 100)
  })
})
