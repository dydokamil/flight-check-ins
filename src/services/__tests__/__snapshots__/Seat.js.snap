// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Seats component should render the component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Seat
    basePrice={0}
    checkInFee={0}
    onBookSeat={[Function]}
    seat={
        Object {
            "id": 1,
          }
    }
    sessionReducer={
        Object {
            "email": "john@doe.com",
            "token": "123abc",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Button
          className="seat"
          color="secondary"
          disabled={true}
          id="Popover1"
          onClick={[Function]}
          tag="button"
>
          1
</Button>,
        <Popover
          delay={
                    Object {
                              "hide": 0,
                              "show": 0,
                            }
          }
          hideArrow={false}
          isOpen={false}
          placement="bottom"
          placementPrefix="bs-popover"
          target="Popover1"
          toggle={[Function]}
>
          <PopoverHeader
                    tag="h3"
          >
                    Seat 
                    1
          </PopoverHeader>
          <PopoverBody
                    tag="div"
          >
                    <div>
                              <h5>
                                        Price: $
                                        NaN
                                         + $
                                        0
                              </h5>
                              <small>
                                        Fees included.
                              </small>
                    </div>
                    <div>
                              <Button
                                        color="success"
                                        disabled={true}
                                        onClick={[Function]}
                                        tag="button"
                              >
                                        Book
                              </Button>
                    </div>
          </PopoverBody>
</Popover>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": 1,
          "className": "seat",
          "color": "secondary",
          "disabled": true,
          "id": "Popover1",
          "onClick": [Function],
          "tag": "button",
        },
        "ref": null,
        "rendered": 1,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <PopoverHeader
              tag="h3"
>
              Seat 
              1
</PopoverHeader>,
            <PopoverBody
              tag="div"
>
              <div>
                            <h5>
                                          Price: $
                                          NaN
                                           + $
                                          0
                            </h5>
                            <small>
                                          Fees included.
                            </small>
              </div>
              <div>
                            <Button
                                          color="success"
                                          disabled={true}
                                          onClick={[Function]}
                                          tag="button"
                            >
                                          Book
                            </Button>
              </div>
</PopoverBody>,
          ],
          "delay": Object {
            "hide": 0,
            "show": 0,
          },
          "hideArrow": false,
          "isOpen": false,
          "placement": "bottom",
          "placementPrefix": "bs-popover",
          "target": "Popover1",
          "toggle": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                "Seat ",
                1,
              ],
              "tag": "h3",
            },
            "ref": null,
            "rendered": Array [
              "Seat ",
              1,
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <div>
                  <h5>
                                    Price: $
                                    NaN
                                     + $
                                    0
                  </h5>
                  <small>
                                    Fees included.
                  </small>
</div>,
                <div>
                  <Button
                                    color="success"
                                    disabled={true}
                                    onClick={[Function]}
                                    tag="button"
                  >
                                    Book
                  </Button>
</div>,
              ],
              "tag": "div",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h5>
                      Price: $
                      NaN
                       + $
                      0
</h5>,
                    <small>
                      Fees included.
</small>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Price: $",
                        NaN,
                        " + $",
                        0,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Price: $",
                      NaN,
                      " + $",
                      0,
                    ],
                    "type": "h5",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Fees included.",
                    },
                    "ref": null,
                    "rendered": "Fees included.",
                    "type": "small",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Button
                    color="success"
                    disabled={true}
                    onClick={[Function]}
                    tag="button"
>
                    Book
</Button>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Book",
                    "color": "success",
                    "disabled": true,
                    "onClick": [Function],
                    "tag": "button",
                  },
                  "ref": null,
                  "rendered": "Book",
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Button
            className="seat"
            color="secondary"
            disabled={true}
            id="Popover1"
            onClick={[Function]}
            tag="button"
>
            1
</Button>,
          <Popover
            delay={
                        Object {
                                    "hide": 0,
                                    "show": 0,
                                  }
            }
            hideArrow={false}
            isOpen={false}
            placement="bottom"
            placementPrefix="bs-popover"
            target="Popover1"
            toggle={[Function]}
>
            <PopoverHeader
                        tag="h3"
            >
                        Seat 
                        1
            </PopoverHeader>
            <PopoverBody
                        tag="div"
            >
                        <div>
                                    <h5>
                                                Price: $
                                                NaN
                                                 + $
                                                0
                                    </h5>
                                    <small>
                                                Fees included.
                                    </small>
                        </div>
                        <div>
                                    <Button
                                                color="success"
                                                disabled={true}
                                                onClick={[Function]}
                                                tag="button"
                                    >
                                                Book
                                    </Button>
                        </div>
            </PopoverBody>
</Popover>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": 1,
            "className": "seat",
            "color": "secondary",
            "disabled": true,
            "id": "Popover1",
            "onClick": [Function],
            "tag": "button",
          },
          "ref": null,
          "rendered": 1,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <PopoverHeader
                tag="h3"
>
                Seat 
                1
</PopoverHeader>,
              <PopoverBody
                tag="div"
>
                <div>
                                <h5>
                                                Price: $
                                                NaN
                                                 + $
                                                0
                                </h5>
                                <small>
                                                Fees included.
                                </small>
                </div>
                <div>
                                <Button
                                                color="success"
                                                disabled={true}
                                                onClick={[Function]}
                                                tag="button"
                                >
                                                Book
                                </Button>
                </div>
</PopoverBody>,
            ],
            "delay": Object {
              "hide": 0,
              "show": 0,
            },
            "hideArrow": false,
            "isOpen": false,
            "placement": "bottom",
            "placementPrefix": "bs-popover",
            "target": "Popover1",
            "toggle": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  "Seat ",
                  1,
                ],
                "tag": "h3",
              },
              "ref": null,
              "rendered": Array [
                "Seat ",
                1,
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <div>
                    <h5>
                                        Price: $
                                        NaN
                                         + $
                                        0
                    </h5>
                    <small>
                                        Fees included.
                    </small>
</div>,
                  <div>
                    <Button
                                        color="success"
                                        disabled={true}
                                        onClick={[Function]}
                                        tag="button"
                    >
                                        Book
                    </Button>
</div>,
                ],
                "tag": "div",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h5>
                        Price: $
                        NaN
                         + $
                        0
</h5>,
                      <small>
                        Fees included.
</small>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "Price: $",
                          NaN,
                          " + $",
                          0,
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        "Price: $",
                        NaN,
                        " + $",
                        0,
                      ],
                      "type": "h5",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Fees included.",
                      },
                      "ref": null,
                      "rendered": "Fees included.",
                      "type": "small",
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Button
                      color="success"
                      disabled={true}
                      onClick={[Function]}
                      tag="button"
>
                      Book
</Button>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Book",
                      "color": "success",
                      "disabled": true,
                      "onClick": [Function],
                      "tag": "button",
                    },
                    "ref": null,
                    "rendered": "Book",
                    "type": [Function],
                  },
                  "type": "div",
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
